bindings/go/go.mod

```diff
@@ -17,12 +17,22 @@
 
 module opendal.apache.org/go
```

Comment:
Let's use `github.com/apache/opendal/bindings/go` so users can use go binding from git now.

Reply:
I'm not sure whether go support get sub directory.

Reply:
![image](https://github.com/user-attachments/assets/79c15d0d-4b6b-44a2-a993-2ff597b236c4)
Oh it doesn't work.

Reply:
It doesn't work since we are now pointintg to `opendal.apache.org/go`. After we change our mod to `github.com/apache/opendal/bindings/go`. It should work.